src/models/MatrixAction.ts

```diff
@@ -187,9 +187,13 @@ export class MatrixAction {
                 }
 
                 if (filename) {
-                    url += `/${filename}`;
+                    url += `/${encodeURIComponent(filename)}`;
+                    text = `${fileSize} < ${url} >`;
+                }
+                else {
+                    // If not a filename, print the body
+                    text = `${event.content.body} ${fileSize} < ${url} >`;
```

Comment:
What do we expect to be in the body? An accompanying message from the Matrix user?
Do we not post that if there's a filename?

Reply:
The Matrix spec is a bit weird regarding this https://matrix.org/docs/spec/client_server/r0.6.0#m-file

>Required. A human-readable description of the file. This is recommended to be the filename of the original upload.

Which means it could be either a filename or a human-readable description. We have to do a quick regex check above to try to understand which one it is.